vllm/model_executor/guided_decoding.py

```diff
@@ -0,0 +1,91 @@
+from enum import Enum
+from functools import lru_cache
+from json import dumps as json_dumps
+from typing import Union
+from types import SimpleNamespace
+from pydantic import BaseModel
+
+from vllm.entrypoints.openai.protocol import CompletionRequest, ChatCompletionRequest
+try:
+    from outlines.serve.vllm import JSONLogitsProcessor, RegexLogitsProcessor
+except ImportError as e:
+    raise ValueError(
+        "Please install 'outlines' (pip install outlines) to use guided decoding."
+    ) from e
+
+
+def get_guided_decoding_logits_processor(
+        request: Union[CompletionRequest, ChatCompletionRequest],
+        tokenizer
+    ) -> Union[JSONLogitsProcessor, RegexLogitsProcessor]:
+    """
+    Given an OpenAI-compatible request, check for guided decoding parameters
+    and get the necessary logits processor for the given guide.
+    We cache logit processors by (json/regex, tokenizer).
+    """
+
+    if request.guided_json:
```

Comment:
**Suggestion**

I think it would be better to check whether the request contains more than one type of guided decoding parameter and fail instead of defining an implicit ordering.

Reply:
Something like:

```suggestion
    if sum([request.guided_json != None, request.guided_regex != None, request.guided_choice != None]) > 1:
        raise ValueError("You can only use one kind of guided decoding ('json_schema', 'regex' or 'choice').")

    if request.guided_json:
```